src/dali/plugin/input/csv/bitbank_supplemental.py

```diff
@@ -48,22 +51,81 @@ class InputPlugin(AbstractInputPlugin):
+        self.__withdrawals_code: Optional[str] = withdrawals_code
         self.__logger: logging.Logger = create_logger(f"{self.__BITBANK_PLUGIN}/{self.account_holder}")
 
     def load(self) -> List[AbstractTransaction]:
-        return self.parse_withdrawals_file(self.__withdrawals_csv_file)
+        result: List[AbstractTransaction] = []
+
+        if self.__withdrawals_csv_file:
+            result.extend(self.parse_withdrawals_file(self.__withdrawals_csv_file))
+
+        if self.__deposits_csv_file:
+            result.extend(self.parse_deposits_file(self.__deposits_csv_file))
+
+        return result
+
+    def parse_deposits_file(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+        if self.__deposits_code:
+            deposits_code: str = self.__deposits_code
+        else:
+            raise Exception("Bitbank.cc supplemental plugin deposits file declared without deposits code.")
+
+        with open(file_path, encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+            for line in lines:
+                if line[self.__DEPOSIT_STATUS] == "DONE":
+                    raw_data: str = self.__DELIMITER.join(line)
+                    self.__logger.debug("Transaction: %s", raw_data)
+
+                    jst_timezone = PytzTimezone("Asia/Tokyo")
+                    jst_datetime: datetime = jst_timezone.localize(datetime.strptime(line[self.__TIMESTAMP_INDEX], "%Y/%m/%d %H:%M:%S"))
+                    utc_timestamp: str = jst_datetime.astimezone(DatetimeTimezone.utc).strftime("%Y-%m-%d %H:%M:%S%z")
+
+                    result.append(
+                        InTransaction(
+                            plugin=self.__BITBANK_PLUGIN,
+                            unique_id=Keyword.UNKNOWN.value,
+                            raw_data=raw_data,
+                            timestamp=utc_timestamp,
+                            asset=deposits_code,
+                            exchange=self.__BITBANK,
+                            holder=self.account_holder,
+                            transaction_type=Keyword.BUY.value,
+                            spot_price="1.0",
+                            crypto_in=str(line[self.__DEPOSIT_TOTAL]),
+                            crypto_fee=None,
+                            fiat_in_no_fee=str(line[self.__DEPOSIT_TOTAL]),
+                            fiat_in_with_fee=str(line[self.__DEPOSIT_TOTAL]),
+                            fiat_ticker="JPY",
+                            notes="Fiat deposit",
+                        )
+                    )
+
+        return result
 
     def parse_withdrawals_file(self, file_path: str) -> List[AbstractTransaction]:
         result: List[AbstractTransaction] = []
+        if self.__withdrawals_code:
+            withdrawals_code: str = self.__withdrawals_code
```

Comment:
Just a nit, but this new variable seems unnecessary (just access the instance variable)

Reply:
This is stripping the optionality.

Reply:
I meant:
```suggestion
        if not self.__withdrawals_code:
            raise Exception("Bitbank.cc supplemental plugin withdrawals file declared without withdrawals code.")
```

After that you should be able to use `self.__withdrawals_code` without optionality.